packages/commonwealth/client/scripts/views/pages/CommunityManagement/Integrations/Snapshots/validation.ts

```diff
@@ -1,13 +1,10 @@
 import z from 'zod';
 
-const snapshotValidationSchema = z.string().refine(
-  (space) => {
-    const extension = space.slice(space.length - 4);
-    return extension === '.eth' || extension === '.xyz';
-  },
-  {
-    message: 'Snapshot name must be in the form of *.eth or *.xyz',
-  },
-);
+const snapshotValidationSchema = z
+  .string()
+  .refine((space) => /^[a-zA-Z0-9-.]+\.eth$|^[a-zA-Z0-9-]+\.xyz$/.test(space), {
```

Comment:
Minor: there is a missing `.` here for the `.xyz` part of the regex, we need this to allow multiple `.` in the `.xyz` domain

```diff
- /^[a-zA-Z0-9-.]+\.eth$|^[a-zA-Z0-9-]+\.xyz$/
+ /^[a-zA-Z0-9-.]+\.eth$|^[a-zA-Z0-9-.]+\.xyz$/
```

Also as a good convention, we should escape literal hyphen chars, so the regex becomes something like this

```diff
+ /^[a-zA-Z0-9\-.]+\.eth$|^[a-zA-Z0-9\-.]+\.xyz$/
```